pyproject.toml

```diff
@@ -1,43 +1,91 @@
 [build-system]
 requires = [
-    "setuptools>=42",
     "wheel",
     "ninja",
-    "cmake>=3.12",
+    "cmake>=3.20",
     "pybind11>=2.10.4",
-    "numpy",
+    "numpy>=2",
     "PyYAML",
     "nanobind",
+    "delvewheel; platform_system=='Windows'",
+    "setuptools-scm>=8",
+    "scikit-build-core>=0.10,<0.11"
 ]
+build-backend = "scikit_build_core.build"
+
+[project]
+name = "finch-mlir"
+dynamic = ["version"]
+description = "Finch MLIR distribution as wheel."
+readme = "README.md"
+dependencies = [
+    "PyYAML>=6",
+    "numpy>=1.17",
+]
+
+[tool.setuptools_scm]
+
+[tool.scikit-build]
```

Comment:
I think this file should be structured a bit:
- Let's have `# LLVM build` comment line with all LLVM related config
- And a separate `# Finch dialect build` line/section.

Reply:
Actually, `scikit-build-core` only supports a one-step "external" projects build, as explained in [this PDF](https://archive.fosdem.org/2023/schedule/event/mlirdialect/attachments/slides/5740/export/events/attachments/mlirdialect/slides/5740/How_to_Build_your_own_MLIR_Dialect.pdf) on page 7.

Reply:
So this is build the same way as before (llvm build separately, and finch-dialect build out-of-tree separately) but with one `cmake` command instead of two?

Reply:
Nope, finch is built as part of the LLVM build via these two CMake flags:

https://github.com/finch-tensor/Finch-mlir/blob/5b9d8b08f23a1c876477d69149692b175ad42897/pyproject.toml#L38-L39

Reply:
The other method should also be supported if needed, but if won't be used for wheel builds.

Reply:
I think the source is separate, and there's just one CMake build. But I'd hesitate to use the word "separately" as the builds are interleaved with one another.